Tasks/Common/webdeployment-common/fileTransformationsUtility.ts

```diff
@@ -25,7 +25,7 @@ export async function fileTransformations(isFolderBasedDeployment: boolean, JSON
         var environmentName = tl.getVariable('Release.EnvironmentName');
         if(tl.osType().match(/^Win/)) {
             var transformConfigs = ["Release.config"];
-            if(environmentName) {
+            if(environmentName && environmentName != 'Release') {
```

Comment:
This might turn out to be a bug fix, in the case if environment name is Release as the transformation will be applied only once now.